### ISSUE TITLE
[Feature Request] Add support for Cloudflare Web Analytics

### ISSUE THREAD

**Issue by miynat** [adityatelange/hugo-PaperMod#386]
**Feature request**

Is it possible to add support for Cloudflare Web Analytics in addition to existing Google Analytics integration?

Cloudflare provides a free analytics feature for websites - https://www.cloudflare.com/web-analytics/
It would be great to have a built-in integration for that web analytics tool.

Thanks in advance.

**Comment by adityatelange** [comment]
We have no external scripts being pulled from any cdn and it's upto the user, what they want to fetch externally. Adding external scripts doesn't align with the goals of this project!

You can still add this as we have extended_head and extended_footer support. Please visit wiki/faq pages to see how to use this.

**Comment by miynat** [comment]
Thanks for the pointer @adityatelange , I think that extended_footer functionality will work for my use case.

That being said, apparently built-in Google Analytics support is coming from the Hugo itself, and not from the theme, as can be seen here -  _hugo/tpl/tplimpl/embedded/templates/google_analytics_async.html_ 
So I've created a feature request on the Hugo project: https://github.com/gohugoio/hugo/issues/8482

**Comment by cb3inco** [comment]
I've deployed my page with this template to Cloudflare pages...it auto-builds with analytics. :)

**Comment by sid-r-singh** [comment]
> I've deployed my page with this template to Cloudflare pages...it auto-builds with analytics. :)

Hey @cb3inco  can you tell what exactly did you do to integrate Cloudflare analytics ? I am as well using Cloudfare pages for hosting.

**Comment by cb3inco** [comment]
@sid-r-singh - sure.

Under 'Pages' -> 'Your Site' -> Settings Tab -> Enable Web Analytics. 

<img width="784" alt="Pages | Account | Cloudflare - Web Performance   Security 2021-05-24 23-10-14" src="https://user-images.githubusercontent.com/14948644/119375674-479df900-bce5-11eb-8624-70975c2de43a.png">

**Comment by sid-r-singh** [comment]
Got it. It works fantastic. But I still have a doubt. What about this:
![image](https://user-images.githubusercontent.com/68782815/119379713-71ebb880-bcdd-11eb-9e28-4e15efc597e5.png)

Don't we have to include the js snippet in our site ?
How is the analytics being collected then without even embedding any script?

**Comment by cb3inco** [comment]
I don't think you need to. I get analytics working without doing any more than what I showed you. I did check my source after my site built and I saw the Javascript in page. Do you?

**Comment by sid-r-singh** [comment]
Yes, you are correct. I am as well seeing that the script is embedded just before <body> ends.

**Comment by miynat** [comment]
Cloudflare Pages could be embedding this tracking script automatically, but in case of using another hosting option like Gitlab Pages this script should be added manually (I used extended_footer support suggested by @adityatelange ).
Hence I opened this FR about making this a built-in feature. As it turns out afterwards, this needs to be handled in the Hugo itself, and not in this template. That's why I've opened https://github.com/gohugoio/hugo/issues/8482
